rfcs/text/0030/process.yml

```diff
@@ -0,0 +1,182 @@
+---
+- name: process
+  reusable:
+    top_level: true
+    expected:
+      - at: process
+        as: entry_leader
+        short_override: Information about the entry session leader process.
+      - at: process
+        as: session_leader
+        short_override: Information about the session leader process.
+      - at: process
+        as: group_leader
+        short_override: Information about the process group leader process. Undefined if same as process.
+
+  fields:
+
+    - name: ppid
+      level: core
+      type: long
+      description: >
+        Parent process id.
+      example: 4242
+
+    - name: sid
+      level: core
+      type: long
+      description: >
+        Session id.
+      example: 4242
+
+    - name: ppid_start
+      level: extended
+      type: date
+      example: "2016-05-23T08:05:34.853Z"
+      description: >
+        The time this process's parent started.
+
+    - name: pgid_start
+      level: extended
+      type: date
+      example: "2016-05-23T08:05:34.853Z"
+      description: >
+        The time this process's process group started.
+
+    - name: sid_start
+      level: extended
+      type: date
+      example: "2016-05-23T08:05:34.853Z"
+      description: >
+        The time this process's session started.
+
+    - name: previous_args
+      level: extended
+      type: keyword
+      short: Array of process arguments the previous fork or exec.
+      description: >
+        Array of process arguments, starting with the absolute path to the executable.
+
+        May be filtered to protect sensitive information.
+      example: "[\"/usr/bin/ssh\", \"-l\", \"user\", \"10.0.0.16\"]"
+      normalize:
+        - array
+
+    - name: previous_args_count
+      level: extended
+      type: long
+      short: Length of the process.previous_args array.
+      description: >
+        Length of the process.previous_args array.
+
+        This field can be useful for querying or performing bucket analysis on
+        how many arguments were provided to start a process.
+        More arguments may be an indication of suspicious activity.
+      example: 4
+
+    - name: previous_executable
+      level: extended
+      type: keyword
+      description: >
+        Absolute path to the process previous executable.
+      example: /usr/bin/ssh
+      multi_fields:
+        - type: match_only_text
+          name: text
+
+    - name: interactive
+      level: extended
+      type: boolean
+      example: true
+      short: Whether the process is connected to an interactive shell.
+      description: >
+        Whether the process is connected to an interactive shell.
+
+        Process interactivity is infered from the processes file descriptors. If the character device for the controlling tty is the same as stdin and stderr for the process, the process is considered interactive.
+
+        Note: A non-interactive process can belong to an interactive session. Backgrounded processes are an example of this.
```

Comment:
I think the Note should be reworded - a  backgrounded process  (ie on that is not in the foreground process group) should not lose it's interactive status by switching from the foreground to the background.

"Note: a non-interactive process can belong to an interactive session and is simply one does not have open file descriptors reading the controlling TTY on FD 0 (stdin) or writing to the controlling TTY on FD 2 (stderr).   A backgrounded process is still considered interactive it stdin and stderr are connected to the controlling TTY."